share/cmake/modules/FindExtPackages.cmake

```diff
@@ -177,15 +165,22 @@ if(OCIO_BUILD_PYTHON OR OCIO_BUILD_DOCS)
         list(APPEND _Python_COMPONENTS Development)
     endif()
 
+
     # Python
-    find_package(Python ${OCIO_PYTHON_VERSION} REQUIRED
-                 COMPONENTS ${_Python_COMPONENTS})
+    ocio_handle_dependency(  Python REQUIRED
+                             COMPONENTS ${_Python_COMPONENTS}
+                             MIN_VERSION ${OCIO_PYTHON_VERSION}
+                             RECOMMENDED_VERSION ${OCIO_PYTHON_VERSION}
+                             RECOMMENDED_VERSION_REASON "Latest version tested with OCIO")
 
     if(OCIO_BUILD_PYTHON)
         # pybind11
         # https://github.com/pybind/pybind11
         # pybind11 2.9 fixes issues with MS Visual Studio 2022 (Debug).
-        find_package(pybind11 2.9.2 REQUIRED)
+        ocio_handle_dependency(  pybind11 REQUIRED ALLOW_INSTALL
+                                 MIN_VERSION 2.9.2
+                                 RECOMMENDED_VERSION 2.9.2
+                                 RECOMMENDED_VERSION_REASON "Pybind 2.10.0+ does not work with Python 2.7 anymore")
```

Comment:
Note #1770 removes official support for Python 2.7, I'm not sure we have urgent needs to update the `pybind11` recommended version but thought I would just mention it.